modules/aws-http4s/test/src-jvm/smithy4s/aws/internals/AwsSignatureTest.scala

```diff
@@ -89,12 +89,77 @@ object AwsSignatureTest extends SimpleIOSuite with Checkers {
       awsRequest: SdkHttpFullRequest
   )
   object TestInput {
-    implicit val show: Show[TestInput] = Show.fromToString
+    implicit val showContentStreamProvider: Show[ContentStreamProvider] =
+      Show.show { csp =>
+        scala.io.Source.fromInputStream(csp.newStream()).mkString
+      }
+
+    implicit val showSdkHttpFullRequest: Show[SdkHttpFullRequest] = Show.show {
+      (req: SdkHttpFullRequest) =>
+        val headers = Option(req.headers().asScala.toList)
+          .filter(_.nonEmpty)
+          .map(_.map { case (k, v) => s"$k: $v" })
+          .map(
+            _.mkString("\n        ", System.lineSeparator() + "        ", "")
+          )
+          .getOrElse(" none")
+
+        s"""(SdkHttpFullRequest)
+           |    - protocol: ${req.protocol()}
+           |    - host: ${req.host()}
+           |    - port: ${req.port()}
+           |    - method: ${req.method()}
+           |    - encodedPath: ${req.encodedPath()}
+           |    - headers:$headers
+           |    - contentStreamProvider: ${req.contentStreamProvider.toScala.show}
+           |""".stripMargin
+    }
+
+    implicit val showTestInput: Show[TestInput] = Show.show { testInput =>
+      import testInput._
+      s"""TestInput:
+         |  - serviceName: $serviceName
+         |  - operationName: $operationName
+         |  - smithy4sTimestamp: $smithy4sTimestamp
+         |  - smithy4sCredentials: $smithy4sCredentials
+         |  - smithy4sRegion: $smithy4sRegion
+         |  - awsRequest: ${awsRequest.show}
+         |""".stripMargin
+    }
 
-    val genAwsRequest = for {
+    val genPathSegment: Gen[String] =
+      Gen
+        .frequency(
+          1 -> Gen.oneOf("..", "."),
```

Comment:
Path normalization should remove these when it's enabled, so these dot segments are sometimes added to the path to make sure that happens.